src/main/scala/Mathematics/PrimeFactors.scala

```diff
@@ -0,0 +1,19 @@
+package Mathematics
+
+object PrimeFactors {
+
+	/**
+	    * Method returns list of prime factors of number n
+	    *
+	    * @param num
+    	    * @return
+    	*/
+
+	def primeFactors(number: Long): List[Long] = {
+	    val exists = (2L to math.sqrt(number).toLong).find(number % _ == 0)
+	    exists match {
+      		case Some(factor) => factor :: primeFactors(number/factor)
+      		case None => List(number)
```

Comment:
You can move this `case` before the one above. This way you will have the recursive call in tail position which is the safe way of doing recursion.
See: https://www.scala-exercises.org/scala_tutorial/tail_recursion

Reply:
done

Reply:
done.
Thank You for sharing this article.
As I am learning Scala so, this exercise and the other tutorials connected to this helped me more in learning.